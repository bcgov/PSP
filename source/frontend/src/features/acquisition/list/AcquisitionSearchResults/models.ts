import { Api_AcquisitionFile, Api_AcquisitionFileProperty } from 'models/api/AcquisitionFile';
import { Api_Project } from 'models/api/Project';
import Api_TypeCode from 'models/api/TypeCode';

export class AcquisitionSearchResultModel {
  id?: number;
  fileNumber?: string;
  fileName?: string;
  regionCode?: string;
  appLastUpdateUserid?: string;
  appCreateTimestamp?: string;
  appCreateUserid?: string;
  appLastUpdateTimestamp?: string;
  acquisitionFileStatusTypeCode?: Api_TypeCode<string>;
  fileProperties?: Api_AcquisitionFileProperty[];
  project?: Api_Project;

  static fromApi(base: Api_AcquisitionFile): AcquisitionSearchResultModel {
    var newModel = new AcquisitionSearchResultModel();
    newModel.id = base.id;
    newModel.fileName = base.fileName;
    newModel.fileNumber = base.fileNumber;
    newModel.regionCode = base.regionCode?.description;
    newModel.appLastUpdateUserid = base.appLastUpdateUserid;
    newModel.appCreateTimestamp = base.appCreateTimestamp;
    newModel.appCreateUserid = base.appCreateUserid;
    newModel.appLastUpdateTimestamp = base.appLastUpdateTimestamp;
    newModel.acquisitionFileStatusTypeCode = base.fileStatusTypeCode;
    newModel.fileProperties = base.fileProperties;
    newModel.project = base.project;
    return newModel;
  }
}
