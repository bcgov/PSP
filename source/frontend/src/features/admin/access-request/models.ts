import { ContactMethodTypes } from '@/constants/contactMethodType';
import { UserTypes } from '@/constants/index';
import { ApiGen_Concepts_AccessRequest } from '@/models/api/generated/ApiGen_Concepts_AccessRequest';
import { UtcIsoDateTime } from '@/models/api/UtcIsoDateTime';
import { getEmptyBaseAudit } from '@/models/defaultInitializers';
import { NumberFieldValue } from '@/typings/NumberFieldValue';
import { getPreferredContactMethodValue } from '@/utils/contactMethodUtil';
import {
  fromTypeCode,
  stringToNumber,
  stringToNumberOrNull,
  toTypeCodeNullable,
} from '@/utils/formUtils';

export class FormAccessRequest {
  public id: NumberFieldValue;
  public userId: NumberFieldValue;
  public note: string;
  public roleName: string;
  public roleId: NumberFieldValue;
  public accessRequestStatusTypeCodeId: string;
  public accessRequestStatusTypeCodeName: string;
  public regionCodeId: NumberFieldValue;
  public regionName: string;
  public appCreateTimestamp?: UtcIsoDateTime;
  public email: string;
  public firstName: string;
  public surname: string;
  public middleNames: string;
  public position: string;
  public userTypeCode: string;
  public businessIdentifierValue: string;
  public keycloakUserGuid: string;
  public rowVersion?: number;

  constructor(accessRequest: ApiGen_Concepts_AccessRequest) {
    this.id = accessRequest.id ?? '';
    this.userId = accessRequest.userId ?? '';
    this.roleId = accessRequest.roleId ?? '';
    this.roleName = accessRequest?.role?.name ?? '';
    this.note = accessRequest.note ?? '';
    this.accessRequestStatusTypeCodeId = accessRequest.accessRequestStatusTypeCode?.id ?? '';
    this.accessRequestStatusTypeCodeName = accessRequest.accessRequestStatusTypeCode?.id ?? '';
    this.regionCodeId = accessRequest.regionCode?.id ?? '';
    this.regionName = accessRequest.regionCode?.description ?? '';
    this.email =
      (accessRequest?.user?.person?.contactMethods &&
        getPreferredContactMethodValue(
          accessRequest?.user?.person?.contactMethods,
          ContactMethodTypes.WorkEmail,
          ContactMethodTypes.PersonalEmail,
        )) ??
      '';
    this.firstName = accessRequest?.user?.person?.firstName ?? '';
    this.surname = accessRequest?.user?.person?.surname ?? '';
    this.middleNames = accessRequest?.user?.person?.middleNames ?? '';
    this.businessIdentifierValue = accessRequest.user?.businessIdentifierValue ?? '';
    this.keycloakUserGuid = accessRequest?.user?.guidIdentifierValue ?? '';
    this.position = accessRequest?.user?.position ?? '';
    this.userTypeCode = fromTypeCode(accessRequest?.user?.userTypeCode) ?? UserTypes.Contractor;
    this.rowVersion = accessRequest.rowVersion ?? undefined;
  }

  public toApi(): ApiGen_Concepts_AccessRequest {
    return {
      id: stringToNumber(this.id),
      userId: stringToNumberOrNull(this.userId),
      roleId: stringToNumberOrNull(this.roleId),
      note: this.note,
      accessRequestStatusTypeCode: toTypeCodeNullable(this.accessRequestStatusTypeCodeId),
      regionCode: this.regionCodeId ? toTypeCodeNullable<number>(this.regionCodeId) : null,
      user: {
        guidIdentifierValue: this.keycloakUserGuid,
        position: this.position,
        userTypeCode: toTypeCodeNullable(this.userTypeCode),
        userRoles: [],
        userRegions: [],
        approvedById: 0,
        businessIdentifierValue: null,
        id: 0,
        isDisabled: false,
        issueDate: null,
        lastLogin: null,
        note: null,
        person: null,
        ...getEmptyBaseAudit(),
      },
      role: null,
      ...getEmptyBaseAudit(this.rowVersion),
    };
  }
}

export const userTypeCodeValues = [
  {
    radioValue: UserTypes.MinistryStaff,
    radioLabel: 'Ministry staff',
  },
  {
    radioValue: UserTypes.Contractor,
    radioLabel: 'Contractor',
  },
];
