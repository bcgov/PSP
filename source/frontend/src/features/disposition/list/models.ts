import isNumber from 'lodash/isNumber';

import { SelectOption } from '@/components/common/form';
import { Api_DispositionFilter } from '@/models/api/DispositionFilter';
import { ApiGen_Concepts_DispositionFile } from '@/models/api/generated/ApiGen_Concepts_DispositionFile';
import { ApiGen_Concepts_DispositionFileProperty } from '@/models/api/generated/ApiGen_Concepts_DispositionFileProperty';
import { ApiGen_Concepts_DispositionFileTeam } from '@/models/api/generated/ApiGen_Concepts_DispositionFileTeam';

export class DispositionFilterModel {
  searchBy: string = 'address';
  pin: string = '';
  pid: string = '';
  address: string = '';
  dispositionTeamMember: SelectOption | null = null;
  fileNameOrNumberOrReference: string = '';
  dispositionFileStatusCode: string = '';
  dispositionStatusCode: string = '';
  dispositionTypeCode: string = '';

  toApi(): Api_DispositionFilter {
    const personMemberId =
      this.dispositionTeamMember?.value && String(this.dispositionTeamMember.value).startsWith('P-')
        ? String(this.dispositionTeamMember.value).substring(2)
        : null;
    const orgMemberId =
      this.dispositionTeamMember?.value && String(this.dispositionTeamMember.value).startsWith('O-')
        ? String(this.dispositionTeamMember.value).substring(2)
        : null;

    return {
      searchBy: this.searchBy,
      pin: this.pin,
      pid: this.pid,
      address: this.address,
      fileNameOrNumberOrReference: this.fileNameOrNumberOrReference,
      dispositionFileStatusCode: this.dispositionFileStatusCode,
      dispositionStatusCode: this.dispositionStatusCode,
      dispositionTypeCode: this.dispositionTypeCode,
      // disposition team members
      teamMemberPersonId:
        personMemberId && isNumber(+personMemberId) ? Number(personMemberId) : null,
      teamMemberOrganizationId: orgMemberId && isNumber(+orgMemberId) ? Number(orgMemberId) : null,
    };
  }

  static fromApi(
    base: Api_DispositionFilter,
    teamMemberOptions: SelectOption[] = [],
  ): DispositionFilterModel {
    const newModel = new DispositionFilterModel();
    newModel.searchBy = base.searchBy ?? 'address';
    newModel.pin = base.pin ?? '';
    newModel.pid = base.pid ?? '';
    newModel.address = base.address ?? '';
    newModel.fileNameOrNumberOrReference = base.fileNameOrNumberOrReference ?? '';
    newModel.dispositionFileStatusCode = base.dispositionFileStatusCode ?? '';
    newModel.dispositionStatusCode = base.dispositionStatusCode ?? '';
    newModel.dispositionTypeCode = base.dispositionTypeCode ?? '';
    // disposition team members
    newModel.dispositionTeamMember = base.teamMemberPersonId
      ? teamMemberOptions.find(c => c.value === `P-${base.teamMemberPersonId}`) ?? null
      : base.teamMemberOrganizationId
      ? teamMemberOptions.find(c => c.value === `O-${base.teamMemberOrganizationId}`) ?? null
      : null;

    return newModel;
  }
}

export class DispositionSearchResultModel {
  id: number | null = null;
  fileNumber: string = '';
  fileName: string = '';
  fileReference: string = '';
  region: string = '';
  dispositionTypeCode: string = '';
  dispositionStatusTypeCode: string = '';
  dispositionFileStatusTypeCode: string = '';
  dispositionTeam: ApiGen_Concepts_DispositionFileTeam[] = [];
  fileProperties?: ApiGen_Concepts_DispositionFileProperty[] = [];

  static fromApi(base: ApiGen_Concepts_DispositionFile): DispositionSearchResultModel {
    var newModel = new DispositionSearchResultModel();
    newModel.id = base.id ?? null;
    newModel.fileNumber = base.fileNumber ?? '';
    newModel.fileName = base.fileName ?? '';
    newModel.fileReference = base.fileReference ?? '';
    newModel.region = base.regionCode?.description ?? '';
    newModel.dispositionTypeCode = base.dispositionTypeCode?.description ?? '';
    newModel.dispositionStatusTypeCode = base.dispositionStatusTypeCode?.description ?? '';
    newModel.dispositionFileStatusTypeCode = base.fileStatusTypeCode?.description ?? '';
    newModel.dispositionTeam = base.dispositionTeam || [];
    newModel.fileProperties = base.fileProperties || [];

    return newModel;
  }
}
