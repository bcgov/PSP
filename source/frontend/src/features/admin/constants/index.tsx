import { ColumnWithProps } from 'components/Table';
import { IOrganization, IOrganizationRecord } from 'interfaces';
import * as React from 'react';
import { Link } from 'react-router-dom';
import { CellProps } from 'react-table';

export const columnDefinitions: ColumnWithProps<IOrganizationRecord>[] = [
  {
    Header: 'Organization name',
    accessor: 'name',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Short name',
    accessor: 'code',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Description',
    accessor: 'description',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Parent Organization',
    accessor: 'parent',
    align: 'left',
    clickable: true,
    Cell: (props: CellProps<IOrganization>) => {
      return (
        <Link to={`/admin/organization/${props.row.original.parentId}`}>
          {props.row.original.parent}
        </Link>
      );
    },
  },
];
