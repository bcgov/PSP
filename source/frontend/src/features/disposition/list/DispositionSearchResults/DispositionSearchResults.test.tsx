import { Claims } from '@/constants/claims';
import { getEmptyPerson } from '@/mocks/contacts.mock';
import { mockDispositionFileResponse } from '@/mocks/dispositionFiles.mock';
import { mockLookups } from '@/mocks/lookups.mock';
import { getEmptyOrganization } from '@/mocks/organization.mock';
import { getMockApiProperty } from '@/mocks/properties.mock';
import { ApiGen_Concepts_DispositionFile } from '@/models/api/generated/ApiGen_Concepts_DispositionFile';
import { lookupCodesSlice } from '@/store/slices/lookupCodes';
import { render, RenderOptions, screen } from '@/utils/test-utils';

import { DispositionSearchResultModel } from '../models';
import {
  DispositionSearchResults,
  IDispositionSearchResultsProps,
} from './DispositionSearchResults';

const setSort = vi.fn();
const mockResults: ApiGen_Concepts_DispositionFile[] = [
  mockDispositionFileResponse(1, 'test disposition'),
];

describe('Disposition search results table', () => {
  const getTableRows = () => document.querySelectorAll('.table .tbody .tr-wrapper');

  const setup = (
    renderOptions: RenderOptions & Partial<IDispositionSearchResultsProps> = { results: [] },
  ) => {
    const { results, ...rest } = renderOptions;
    const utils = render(<DispositionSearchResults results={results ?? []} setSort={setSort} />, {
      store: {
        [lookupCodesSlice.name]: { lookupCodes: mockLookups },
      },
      claims: [Claims.DISPOSITION_VIEW],
      ...rest,
    });

    return { ...utils };
  };

  beforeEach(() => {
    setSort.mockClear();
  });

  it('matches snapshot', async () => {
    const { asFragment } = setup({
      results: mockResults.map(a => DispositionSearchResultModel.fromApi(a)),
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays a message when no matching records found', async () => {
    setup({ results: [] });
    const message = await screen.findByText(
      'No matching results can be found. Try widening your search criteria.',
    );
    expect(message).toBeVisible();
    expect(getTableRows().length).toBe(0);
  });

  it('displays disposition file name', async () => {
    setup({ results: mockResults.map(a => DispositionSearchResultModel.fromApi(a)) });
    const text = await screen.findByText(mockResults[0].fileName!);
    expect(text).toBeVisible();
  });

  it('displays disposition file number', async () => {
    setup({ results: mockResults.map(a => DispositionSearchResultModel.fromApi(a)) });
    const text = await screen.findByText(`D-${mockResults[0].fileNumber!}`);
    expect(text).toBeVisible();
  });

  it('displays disposition legacy file reference', async () => {
    setup({ results: mockResults.map(a => DispositionSearchResultModel.fromApi(a)) });
    const text = await screen.findByText(mockResults[0].fileReference!);
    expect(text).toBeVisible();
  });

  it('displays multiple file properties', () => {
    setup({
      results: [
        DispositionSearchResultModel.fromApi({
          ...mockDispositionFileResponse(),
          fileProperties: [
            {
              id: 100,
              fileId: 1,
              propertyId: 1,
              property: { ...getMockApiProperty(), id: 1 },
              displayOrder: null,
              file: null,
              propertyName: null,
              location: null,
              rowVersion: null,
            },
            {
              id: 200,
              fileId: 1,
              propertyId: 2,
              property: { ...getMockApiProperty(), id: 2 },
              displayOrder: null,
              file: null,
              propertyName: null,
              location: null,
              rowVersion: null,
            },
            {
              id: 300,
              fileId: 1,
              propertyId: 3,
              property: { ...getMockApiProperty(), id: 3 },
              displayOrder: null,
              file: null,
              propertyName: null,
              location: null,
              rowVersion: null,
            },
          ],
        }),
      ],
    });

    const addresses = screen.getAllByText('1234 mock Street', { exact: false });
    expect(addresses).toHaveLength(2);
    expect(screen.getAllByText('[+1 more...]')).toHaveLength(1);
  });

  it('displays a team member organization', () => {
    setup({
      results: [
        DispositionSearchResultModel.fromApi({
          ...mockDispositionFileResponse(),
          dispositionTeam: [
            {
              id: 1,
              dispositionFileId: 1,
              organizationId: 6,
              organization: {
                ...getEmptyOrganization(),
                id: 6,
                isDisabled: false,
                name: 'FORTIS BC',
                alias: 'FORTIS',
                incorporationNumber: '123456789',
                organizationPersons: [],
                organizationAddresses: [],
                contactMethods: [],
                comment: '',
                rowVersion: 1,
              },
              teamProfileType: {
                id: 'MAJORPRJ',
                description: 'Major Projects',
                isDisabled: false,
                displayOrder: null,
              },
              primaryContact: null,
              primaryContactId: null,
              rowVersion: null,
              teamProfileTypeCode: null,
              person: null,
              personId: null,
            },
          ],
        }),
      ],
    });

    expect(screen.getByText('FORTIS BC (Major Projects)')).toBeVisible();
  });

  it('displays a team member person', () => {
    setup({
      results: [
        DispositionSearchResultModel.fromApi({
          ...mockDispositionFileResponse(),
          dispositionTeam: [
            {
              id: 1,
              dispositionFileId: 1,
              personId: 6,
              person: {
                ...getEmptyPerson(),
                id: 6,
                isDisabled: false,
                rowVersion: 1,
                firstName: 'chester',
                surname: 'tester',
                comment: null,
                contactMethods: null,
                middleNames: null,
                personAddresses: null,
                personOrganizations: null,
                preferredName: null,
              },
              teamProfileType: {
                id: 'MoTIReg',
                description: 'MoTI Region',
                isDisabled: false,
                displayOrder: null,
              },
              primaryContact: null,
              primaryContactId: null,
              rowVersion: null,
              teamProfileTypeCode: null,
              organization: null,
              organizationId: null,
            },
          ],
        }),
      ],
    });

    expect(screen.getByText('chester tester (MoTI Region)')).toBeVisible();
  });

  it('displays multiple team members', () => {
    setup({
      results: [
        DispositionSearchResultModel.fromApi({
          ...mockDispositionFileResponse(),
          dispositionTeam: [
            {
              id: 1,
              dispositionFileId: 1,
              organizationId: 6,
              organization: {
                ...getEmptyOrganization(),
                id: 6,
                isDisabled: false,
                name: 'FORTIS BC',
                alias: 'FORTIS',
                incorporationNumber: '123456789',
                organizationPersons: [],
                organizationAddresses: [],
                contactMethods: [],
                comment: '',
                rowVersion: 1,
              },
              teamProfileType: {
                id: 'MAJORPRJ',
                description: 'Major Projects',
                isDisabled: false,
                displayOrder: null,
              },
              person: null,
              personId: null,
              primaryContact: null,
              primaryContactId: null,
              rowVersion: null,
              teamProfileTypeCode: null,
            },
            {
              id: 2,
              dispositionFileId: 1,
              personId: 6,
              person: {
                ...getEmptyPerson(),
                id: 6,
                isDisabled: false,
                rowVersion: 1,
                firstName: 'chester',
                surname: 'tester',
                comment: null,
                contactMethods: null,
                middleNames: null,
                personAddresses: null,
                personOrganizations: null,
                preferredName: null,
              },
              teamProfileType: {
                id: 'MoTIReg',
                description: 'MoTI Region',
                isDisabled: false,
                displayOrder: null,
              },
              primaryContact: null,
              primaryContactId: null,
              rowVersion: null,
              teamProfileTypeCode: null,
              organization: null,
              organizationId: null,
            },
            {
              id: 3,
              dispositionFileId: 1,
              personId: 7,
              person: {
                ...getEmptyPerson(),
                id: 7,
                isDisabled: false,
                rowVersion: 1,
                firstName: 'john',
                surname: 'doe',
                comment: null,
                contactMethods: null,
                middleNames: null,
                personAddresses: null,
                personOrganizations: null,
                preferredName: null,
              },
              teamProfileType: {
                id: 'CAPPROG',
                description: 'Capital Program',
                isDisabled: false,
                displayOrder: null,
              },
              primaryContact: null,
              primaryContactId: null,
              rowVersion: null,
              teamProfileTypeCode: null,
              organization: null,
              organizationId: null,
            },
          ],
        }),
      ],
    });

    expect(screen.getByText('chester tester (MoTI Region),')).toBeVisible();
    expect(screen.getByText('john doe (Capital Program),')).toBeVisible();
    expect(screen.getByText('[+1 more...]')).toBeVisible();
  });
});
