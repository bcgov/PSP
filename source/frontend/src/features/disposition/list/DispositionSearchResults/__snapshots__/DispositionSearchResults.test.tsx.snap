// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Disposition search results table > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
  <div />
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 0.5rem;
}

.c2 {
  width: 1.6rem;
  height: 1.6rem;
}

.c1 {
  width: 1.6rem;
  height: 1.6rem;
}

<div
    class="table"
    data-testid="dispositionFilesTable"
    role="table"
  >
    <div
      class="thead thead-light"
    >
      <div
        class="tr"
        role="row"
        style="display: flex; flex: 1 0 auto; min-width: 270px;"
      >
        <div
          class="th"
          colspan="1"
          role="columnheader"
          style="box-sizing: border-box; flex: 20 0 auto; min-width: 30px; width: 20px; justify-content: right; text-align: right; flex-wrap: wrap; align-items: center; display: flex;"
        >
          <div
            class="sortable-column"
          >
            Disposition file #
            <div
              class="c0"
              data-testid="sort-column-fileNumber"
            >
              <svg
                class="c1"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-bottom: -8px;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128 320l128-128 128 128z"
                />
              </svg>
              <svg
                class="c2"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128 192l128 128 128-128z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="th"
          colspan="1"
          role="columnheader"
          style="box-sizing: border-box; flex: 20 0 auto; min-width: 30px; width: 20px; justify-content: right; text-align: right; flex-wrap: wrap; align-items: center; display: flex;"
        >
          <div
            class="sortable-column"
          >
            Reference #
            <div
              class="c0"
              data-testid="sort-column-fileReference"
            >
              <svg
                class="c1"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-bottom: -8px;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128 320l128-128 128 128z"
                />
              </svg>
              <svg
                class="c2"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128 192l128 128 128-128z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="th"
          colspan="1"
          role="columnheader"
          style="box-sizing: border-box; flex: 30 0 auto; min-width: 30px; width: 30px; justify-content: left; text-align: left; flex-wrap: wrap; align-items: center; display: flex;"
        >
          <div
            class="sortable-column"
          >
            Disposition file name
            <div
              class="c0"
              data-testid="sort-column-fileName"
            >
              <svg
                class="c1"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-bottom: -8px;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128 320l128-128 128 128z"
                />
              </svg>
              <svg
                class="c2"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128 192l128 128 128-128z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="th"
          colspan="1"
          role="columnheader"
          style="box-sizing: border-box; flex: 30 0 auto; min-width: 30px; width: 30px; justify-content: left; text-align: left; flex-wrap: wrap; align-items: center; display: flex;"
        >
          <div
            class="sortable-column"
          >
            Disposition type
            <div
              class="c0"
              data-testid="sort-column-dispositionTypeCode"
            >
              <svg
                class="c1"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-bottom: -8px;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128 320l128-128 128 128z"
                />
              </svg>
              <svg
                class="c2"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128 192l128 128 128-128z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="th"
          colspan="1"
          role="columnheader"
          style="box-sizing: border-box; flex: 20 0 auto; min-width: 30px; width: 20px; justify-content: left; text-align: left; flex-wrap: wrap; align-items: center; display: flex;"
        >
          <div
            class="sortable-column"
          >
            MOTI Region
          </div>
        </div>
        <div
          class="th"
          colspan="1"
          role="columnheader"
          style="box-sizing: border-box; flex: 40 0 auto; min-width: 30px; width: 40px; justify-content: left; text-align: left; flex-wrap: wrap; align-items: center; display: flex;"
        >
          <div
            class="sortable-column"
          >
            Team member
          </div>
        </div>
        <div
          class="th"
          colspan="1"
          role="columnheader"
          style="box-sizing: border-box; flex: 100 0 auto; min-width: 30px; width: 100px; justify-content: left; text-align: left; flex-wrap: wrap; align-items: center; display: flex;"
        >
          <div
            class="sortable-column"
          >
            Civic Address / PID / PIN
          </div>
        </div>
        <div
          class="th"
          colspan="1"
          role="columnheader"
          style="box-sizing: border-box; flex: 20 0 auto; min-width: 30px; width: 20px; justify-content: left; text-align: left; flex-wrap: wrap; align-items: center; display: flex;"
        >
          <div
            class="sortable-column"
          >
            Disposition status
            <div
              class="c0"
              data-testid="sort-column-dispositionStatusTypeCode"
            >
              <svg
                class="c1"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-bottom: -8px;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128 320l128-128 128 128z"
                />
              </svg>
              <svg
                class="c2"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128 192l128 128 128-128z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="th"
          colspan="1"
          role="columnheader"
          style="box-sizing: border-box; flex: 20 0 auto; min-width: 30px; width: 20px; justify-content: left; text-align: left; flex-wrap: wrap; align-items: center; display: flex;"
        >
          <div
            class="sortable-column"
          >
            Status
            <div
              class="c0"
              data-testid="sort-column-dispositionFileStatusTypeCode"
            >
              <svg
                class="c1"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-bottom: -8px;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128 320l128-128 128 128z"
                />
              </svg>
              <svg
                class="c2"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128 192l128 128 128-128z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tbody"
      role="rowgroup"
    >
      <div
        class="tr-wrapper"
      >
        <div
          class="tr"
          role="row"
          style="display: flex; flex: 1 0 auto; min-width: 270px;"
        >
          <div
            class="td clickable"
            role="cell"
            style="box-sizing: border-box; flex: 20 0 auto; min-width: 30px; width: 20px; justify-content: right; text-align: right; flex-wrap: wrap; align-items: center; display: flex;"
            title=""
          >
            <a
              href="/mapview/sidebar/disposition/1"
            >
              D-FILE_NUMBER 3A8F46B
            </a>
          </div>
          <div
            class="td clickable"
            role="cell"
            style="box-sizing: border-box; flex: 20 0 auto; min-width: 30px; width: 20px; justify-content: right; text-align: right; flex-wrap: wrap; align-items: center; display: flex;"
            title=""
          >
            FILE_REFERENCE 8128827 3EAD56A
          </div>
          <div
            class="td clickable"
            role="cell"
            style="box-sizing: border-box; flex: 30 0 auto; min-width: 30px; width: 30px; justify-content: left; text-align: left; flex-wrap: wrap; align-items: center; display: flex;"
            title=""
          >
            test disposition
          </div>
          <div
            class="td clickable"
            role="cell"
            style="box-sizing: border-box; flex: 30 0 auto; min-width: 30px; width: 30px; justify-content: left; text-align: left; flex-wrap: wrap; align-items: center; display: flex;"
            title=""
          >
            Road Closure
          </div>
          <div
            class="td"
            role="cell"
            style="box-sizing: border-box; flex: 20 0 auto; min-width: 30px; width: 20px; justify-content: left; text-align: left; flex-wrap: wrap; align-items: center; display: flex;"
            title=""
          >
            Cannot determine
          </div>
          <div
            class="td clickable"
            role="cell"
            style="box-sizing: border-box; flex: 40 0 auto; min-width: 30px; width: 40px; justify-content: left; text-align: left; flex-wrap: wrap; align-items: center; display: flex;"
            title=""
          >
            <span>
              Alejandro Sanchez (MoTI Solicitor), 
            </span>
            <span>
              Aman Monga (Negotiation agent)
            </span>
          </div>
          <div
            class="td"
            role="cell"
            style="box-sizing: border-box; flex: 100 0 auto; min-width: 30px; width: 100px; justify-content: left; text-align: left; flex-wrap: wrap; align-items: center; display: flex;"
            title=""
          >
            <div
              class="w-100"
            />
          </div>
          <div
            class="td clickable"
            role="cell"
            style="box-sizing: border-box; flex: 20 0 auto; min-width: 30px; width: 20px; justify-content: left; text-align: left; flex-wrap: wrap; align-items: center; display: flex;"
            title=""
          />
          <div
            class="td clickable"
            role="cell"
            style="box-sizing: border-box; flex: 20 0 auto; min-width: 30px; width: 20px; justify-content: left; text-align: left; flex-wrap: wrap; align-items: center; display: flex;"
            title=""
          >
            Draft
          </div>
        </div>
      </div>
    </div>
  </div>
  .c0 {
  margin-top: 0.3rem;
}

.c1 {
  min-width: 5rem;
  max-width: 5rem;
  margin-left: 1rem;
  margin-right: 1rem;
  text-align: center;
  padding: 0;
}

.c1:disabled {
  background: white;
}

<div
    class="row"
  >
    <div
      class="align-self-center col-auto"
    >
      <span>
        1 - 1 of 1
      </span>
    </div>
    <div
      class="ml-auto align-self-center col-auto"
    />
    <div
      class="align-self-center col-auto"
    >
      <div
        class="Menu-root"
      >
        <div
          class="Menu-button"
        >
          <div
            class="d-flex"
          >
            <span
              class="c0"
            >
              Show
            </span>
            <input
              class="c1 form-control"
              type="number"
              value="10"
            />
            <span
              class="c0"
            >
              Entries
            </span>
          </div>
        </div>
        <div
          class="Menu-items"
          style="width: 10.0rem;"
        >
          <div
            class="Menu-options scrollable list-group"
          >
            <div
              class="Menu-item list-group-item"
              title="menu-item-5"
            >
              5
            </div>
            <div
              class="Menu-item list-group-item"
              title="menu-item-10"
            >
              10
            </div>
            <div
              class="Menu-item list-group-item"
              title="menu-item-20"
            >
              20
            </div>
            <div
              class="Menu-item list-group-item"
              title="menu-item-50"
            >
              50
            </div>
            <div
              class="Menu-item list-group-item"
              title="menu-item-100"
            >
              100
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="align-self-center col-auto"
    >
      <ul
        class="pagination"
      >
        <li
          class="page-item disabled"
        >
          <a
            aria-disabled="true"
            aria-label="Previous page"
            class="page-link "
            rel="prev"
            role="button"
            tabindex="-1"
          >
            &lt;
          </a>
        </li>
        <li
          class="page-item active"
        >
          <a
            aria-current="page"
            aria-label="Page 1 is your current page"
            class="page-link"
            rel="canonical"
            role="button"
            tabindex="-1"
          >
            1
          </a>
        </li>
        <li
          class="page-item disabled"
        >
          <a
            aria-disabled="true"
            aria-label="Next page"
            class="page-link "
            rel="next"
            role="button"
            tabindex="-1"
          >
            &gt;
          </a>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
